expression/constant_fold.go

```diff
@@ -29,21 +28,18 @@ func FoldConstant(ctx context.Context, expr Expression) Expression {
 		return expr
 	}
 	args := scalarFunc.GetArgs()
-	datums := make([]types.Datum, 0, len(args))
```

Comment:
Yes, The logic of `Eval` has changed. If all args are constant, they don't need input row.